engine/order_manager.go

```diff
@@ -641,39 +681,109 @@ func (m *OrderManager) processOrders() {
 				if err != nil {
-					log.Error(log.OrderMgr, err)
+					if !errors.Is(err, common.ErrNotYetImplemented) {
+						log.Error(log.OrderMgr, err)
+					}
 				} else {
-					for i := range orders {
-						if orders[i].InternalOrderID != upsertResponse.OrderDetails.InternalOrderID {
+					for z := range openPositions {
+						if len(openPositions[z].Orders) == 0 {
 							continue
 						}
-						orders[i] = orders[len(orders)-1]
-						orders = orders[:len(orders)-1]
+						err = m.processFuturesPositions(exchanges[x], &openPositions[z])
+						if err != nil {
+							log.Errorf(log.OrderMgr, "unable to process future positions for %v %v %v. err: %v", openPositions[z].Exchange, openPositions[z].Asset, openPositions[z].Pair, err)
+						}
 					}
 				}
 			}
-			if !exchanges[i].GetBase().GetSupportedFeatures().RESTCapabilities.GetOrder {
+			if !exchanges[x].GetBase().GetSupportedFeatures().RESTCapabilities.GetOrder {
 				continue
 			}
 			wg.Add(1)
-			go m.processMatchingOrders(exchanges[i], orders, &wg)
+			go m.processMatchingOrders(exchanges[x], orders, &wg)
 		}
 	}
 	wg.Wait()
 }
 
+// processFuturesPositions ensures any open position found is kept up to date in the order manager
+func (m *OrderManager) processFuturesPositions(exch exchange.IBotExchange, position *order.OpenPositionDetails) error {
+	if !m.trackFuturesPositions {
+		return errFuturesTrackingDisabled
+	}
+	if exch == nil {
+		return fmt.Errorf("%w IBotExchange", common.ErrNilPointer)
+	}
+	if position == nil {
+		return fmt.Errorf("%w OpenPositionDetails", common.ErrNilPointer)
+	}
+	if len(position.Orders) == 0 {
+		return fmt.Errorf("%w position for '%v' '%v' '%v' has no orders", errNilOrder, position.Exchange, position.Asset, position.Pair)
+	}
+	var err error
+	for i := range position.Orders {
+		err = m.orderStore.futuresPositionController.TrackNewOrder(&position.Orders[i])
+		if err != nil {
+			return err
+		}
+	}
+	tick, err := exch.FetchTicker(context.TODO(), position.Pair, position.Asset)
+	if err != nil {
+		return fmt.Errorf("%w when fetching ticker data for %v %v %v", err, position.Exchange, position.Asset, position.Pair)
+	}
+	_, err = m.UpdateOpenPositionUnrealisedPNL(position.Exchange, position.Asset, position.Pair, tick.Last, tick.LastUpdated)
+	if err != nil {
+		return fmt.Errorf("%w when updating unrealised PNL for %v %v %v", err, position.Exchange, position.Asset, position.Pair)
+	}
+
+	frp, err := exch.GetFundingRates(context.TODO(), &order.FundingRatesRequest{
+		Asset:                position.Asset,
+		Pairs:                currency.Pairs{position.Pair},
+		StartDate:            position.Orders[0].Date,
+		EndDate:              time.Now(),
+		IncludePayments:      true,
+		IncludePredictedRate: true,
+	})
+	if err != nil {
+		return err
+	}
+	for i := range frp {
+		err = m.orderStore.futuresPositionController.TrackFundingDetails(&frp[i])
+		if err != nil && !errors.Is(err, order.ErrNotPerpetualFuture) {
```

Comment:
Continue what? There can only be one potential funding rate in the slice. This was a check to prevent an error of checking for funding rate details for non-funding rate items.
With the exposure of `IsPerpetualFutureCurrency`, I'll use that to return earlier and remove the errors is